docs/source/API/core/parallel-dispatch/UniqueToken.rst

```diff
@@ -0,0 +1,103 @@
+``UniqueToken``
+===============
+
+Header File: ``Kokkos_Core.hpp``
+
+
+Description
+------------
+
+``UniqueToken`` is a portable way to acquire a unique ID for a thread (``thread-id`` is not portable across execution environments).  The unique ID is scoped by the ``UniqueTokenScope`` template parameter (defaults to ``Instance``, but can be set to ``Global``).
+
+Interface
+---------
+
+.. code-block:: cpp 
+
+    template <class ExecutionSpace = DefaultExecutionSpace, UniqueTokenScope = UniqueTokenScope::Instance> UniqueToken
+
+
+
+Parameters
+-----------
+
+*  ``ExecutionSpace``:  See `Execution Spaces <../execution_spaces.html>`_
+
+*  ``UniqueTokenScope``:  defaults to ``Instance``, and every instance is independent of another.  In contrast, ``Global`` uses one set of unique IDs for all instances.
+
+.. note::
+   In a parallel region, before the main computation, a pool of ``UniqueToken`` (integer) ID is generated, and each ID is released following iteration.
+
+.. warning::
+   ``UniqueToken <ExecutionSpace> token`` *can* be called inside a parallel region, *but* must be released at the end of *each* iteration.
+
+
+
+Constructors
+-------------
+
+  .. code-block:: cpp
+
+     UniqueToken(size_type max_size, ExecutionSpace execution_space = ExecutionSpace())
+     // Scope is instance
+  
+  .. code-block:: cpp
+     
+     UniqueToken(size_type max_size, ExecutionSpace execution_space = ExecutionSpace()) requires(TokenScope == Global); 
+     // Scope is global
+
+
+Public Member Functions
+------------------------
+
+ .. code-block:: cpp
+    
+    UniqueToken(size_type max_size, ExecutionSpace execution_space=ExecutionSpace())
+     
+ .. code-block:: cpp
+    
+    size_type size()
+    // Returns the size of the token pool    
```

Comment:
There is a dedicated syntax for return
```suggestion
   :returns: the size of the token pool    
```

Reply:
`:return:` doesn't work (as a [Sphinx directive](https://www.sphinx-doc.org/en/master/usage/domains/cpp.html)) with `.. code-block:: cpp` or `.. cpp:function:: `

Reply:
https://www.sphinx-doc.org/en/master/usage/domains/cpp.html#info-field-lists

Reply:
Here,  `code-block:: cpp`  (vs. `cpp:function::`) is being used